File: ios/LocalServer.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(LocalServer, NSObject)

RCT_EXTERN_METHOD(multiply:(float)a withB:(float)b
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(startServer:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/LocalServer.swift
@objc(LocalServer)
class LocalServer: NSObject {

    @objc(multiply:withB:withResolver:withRejecter:)
    func multiply(a: Float, b: Float, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
        resolve(a*b)
    }
    
    @objc(startServer:withRejecter:)
    func startServer(resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
        print("startServer - started")
        resolve(true)
    }
}
